File: Python/Two Sum/TwoSum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        result = []
        nums_dict = {}
        for i in range(len(nums)):
            n = nums[i]
            residual = target - n
            if (residual in nums_dict.keys()):
                if (i != nums_dict[residual]):
                    idx1 = i if i < nums_dict[residual] else nums_dict[
                        residual]
                    idx2 = i if i >= nums_dict[residual] else nums_dict[
                        residual]
                    result.append(idx1)
                    result.append(idx2)
                    break
            else:
                nums_dict[n] = i
        return result

File: C#/Two Sum/TwoSum.cs
public class Solution {
    public int[] TwoSum (int[] nums, int target) {
        int[] result = new int[2];
        int num, num2, residual;
        bool found = false;
        for (int i = 0; i < nums.Length - 1; i++) {
            num = nums[i];
            result[0] = i;
            residual = target - num;
            for (int j = i + 1; j < nums.Length; j++) {
                num2 = nums[j];
                if (num2 == residual) {
                    result[1] = j;
                    found = true;
                    break;
                }
            }
            if (found) break;
            if (i == nums.Length - 2) {
                throw new Exception ("No Solution");
            }
        }
        return result;
    }
}